snapcraft/common.py

```diff
@@ -58,8 +62,9 @@ def run_output(cmd, **kwargs):
         f.write('\n')
         f.write('exec $*')
         f.flush()
-        return subprocess.check_output(['/bin/sh', f.name] + cmd,
-                                       **kwargs).decode('utf8').strip()
+        return subprocess.check_output([
+            '/bin/sh', f.name] + cmd, **kwargs).decode(
+                sys.getfilesystemencoding()).strip()
```

Comment:
I went under the assumption that stdout and stderr are also just files. In fairness I was trying to get rid of this error:
`'utf-8' codec can't encode character '\\udcc3' in position 137:\nsurrogates not allowed\n"`

from http://162.213.35.179:8080/job/github-snapcraft-integration-tests-cloud/23/testReport/tests/TestSnapcraftExamples/test_example_java_hello_world_/